src/Microsoft.Health.Dicom.Core.UnitTests/Features/CustomTag/CustomTagCacheTests.cs

```diff
@@ -0,0 +1,98 @@
+            Assert.NotEqual(lastRefreshTime, customTagCache.LastRefreshTime);
+        }
+
+        [Fact]
+        public async Task GivenRefreshedCustomTagCache_WhenGetCustomTagsIsCalled_ShouldNotRefreshCache()
+        {
+            long expectedVersion1 = 1, expectedVersion2 = 2;
+            List<CustomTagEntry> expectedResult1 = new List<CustomTagEntry>() { CustomTagTestHelper.CreateCustomTagEntry(expectedVersion1) };
+            List<CustomTagEntry> expectedResult2 = new List<CustomTagEntry>() { CustomTagTestHelper.CreateCustomTagEntry(expectedVersion2) };
+            ICustomTagStore customTagStore = CreateCustomTagStore(expectedResult1, expectedResult2);
+
+            CustomTagCache customTagCache = new CustomTagCache(customTagStore, TimeSpan.FromSeconds(6));
+            await customTagCache.GetCustomTagsAsync();
+            DateTimeOffset lastRefreshTime = customTagCache.LastRefreshTime.Value;
+            await Task.Delay(TimeSpan.FromSeconds(1));
+
+            CustomTagList customTags2 = await customTagCache.GetCustomTagsAsync();
+            Assert.Equal(expectedResult1, customTags2.CustomTags);
+            Assert.Equal(lastRefreshTime, customTagCache.LastRefreshTime);
+        }
+
+        [Fact]
+        public async Task GivenRefreshedCustomTagCacheWithEmptyList_WhenGetCustomTagsIsCalled_ShouldNotRefreshCache()
+        {
+            long expectedVersion2 = 2;
+            List<CustomTagEntry> expectedResult1 = new List<CustomTagEntry>();
+            List<CustomTagEntry> expectedResult2 = new List<CustomTagEntry>() { CustomTagTestHelper.CreateCustomTagEntry(expectedVersion2) };
+            ICustomTagStore customTagStore = CreateCustomTagStore(expectedResult1, expectedResult2);
+
+            CustomTagCache customTagCache = new CustomTagCache(customTagStore, DefaultExpirationInterval);
+            await customTagCache.GetCustomTagsAsync();
+            DateTimeOffset lastRefreshTime = customTagCache.LastRefreshTime.Value;
+            await Task.Delay(TimeSpan.FromSeconds(1));
+            CustomTagList customTags2 = await customTagCache.GetCustomTagsAsync();
+            Assert.Equal(expectedResult1, customTags2.CustomTags);
+            Assert.Equal(lastRefreshTime, customTagCache.LastRefreshTime);
+        }
+
+        private ICustomTagStore CreateCustomTagStore(List<CustomTagEntry> firstCallReturn, List<CustomTagEntry> secondCallReturn)
+        {
+            ICustomTagStore customTagStore = Substitute.For<ICustomTagStore>();
+            customTagStore.TryRefreshCustomTags(out Arg.Any<CustomTagList>(), default)
+              .ReturnsForAnyArgs(true);
+
+            customTagStore.WhenForAnyArgs(store => store.TryRefreshCustomTags(out Arg.Any<CustomTagList>(), default))
+                .Do(Callback.First(callInfo => callInfo[0] = new CustomTagList(firstCallReturn))
+                .ThenKeepDoing(callInfo => callInfo[0] = new CustomTagList(secondCallReturn))
+                .AndAlways(callInfo => { }));
```

Comment:
What if you don't need to validate multiple calls?

Reply:
Ideally the method should take an array as input, but  unfortunately nsubstitute doesn't accept an array as input, so if we need to create our own version, the code will be looks like:
if(ary.length==1) {}
else if(ary.length==2){}
...

so I use a simpler way: since at most we only use 2 different returns, the method take 2 return as input, and process as the code. If only have 1 return, pass same return for both them should be good.